### ISSUE TITLE
Docusaurus config is confusing: `const config` vs `module.exports` 

### ISSUE THREAD

**Issue by thadguidry** [facebook/docusaurus#8925]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Description

One of the most frustrating parts in the docs is when you install and get Docusaurus running, but cannot find how to properly configure 2 or more themes.  The examples always use `module.exports = { }` but yet the default shipping config of Docusaurus uses an object with simply `config`.  So when someone wants to simply add let's say Mermaid theme for their markdown pages, it's super confusing how to simply add Mermaid as a theme into the docs, or blog sections.

I would love to improve the docs, but need the answer before I can contribute.
I wanted to do something as easy as this:
```
module.exports = {
  config,
  themes: ['@docusaurus/theme-mermaid'],
}
```
but of course that syntax isn't respected.

I did add the other config parts into the preset-classic ThemeConfig section
```
      // MERMAID CONFIG //
      markdown: {
        mermaid: true,
      },

      mermaid: {
        options: {
          maxTextSize: 50,
        }
      }
      // END OF MERMAID CONFIG //
```
but of course the pages do not show mermaid styling, but only the raw codeblock text of the mermaid graph.
I suspected I needed the mermaid theme still to be added, but on the preset-classic, not sure where or how.
In general, how to use the default installation to add or use 2 themes.  Because `module.exports = config` is not easily understood other than it's a Config type.

### Self-service

- [x] I'd be willing to address this documentation request myself.

**Comment by Josh-Cena** [comment]
Maybe we can try to improve here, but we can't really spare teaching the basics of Node and JS.

This:

```js
const config = {...};

module.exports = config;
```

Is equivalent to:

```js
module.exports = {...};
```

Therefore all you should care is what comes between the curly braces. The docs tell you to add two fields called `markdown` and `thèmes` (neither of which you should have initializing the new project).

**Comment by thadguidry** [comment]
@Josh-Cena I got it working, and it was something more simple ... I removed the config constant and only wrapped with `module.exports = { }` as you were trying to explain.  For some reason, I thought making the change below would break config again, which I think I tried but had a red herring error message because of wrongly placed mermaid config.  Anyways, got it working as you were trying to lead me into that basic direction.  So then a question arrives...

So I'm curious if all the examples and documentation say `module.exports = ` then why confuse initial users by having the first default config to even have this:

```
/** @type {import('@docusaurus/types').Config} */
const config = {
  title: 'Welcome to ...',
  tagline: 'A low-code API development platform',
  favicon: 'img/favicon.ico',
  etc...
```  

rather than having this ? :

```
/** @type {import('@docusaurus/types').Config} */
module.exports = {
  title: 'Welcome to HOMI',
  tagline: 'A low-code API development platform',
  favicon: 'img/favicon.ico',
  etc...
```

**Comment by slorber** [comment]
I used the `const config` constant because it was needed for my IDE and TS to understand the line just above: `/** @type {import('@docusaurus/types').Config} */`

This line permits to tell your IDE the shape of the config object, provides autocompletion features and TypeScript typechecking to ensure you only use valid config.

See for example:

<img width="400" alt="CleanShot 2023-04-27 at 12 17 21@2x" src="https://user-images.githubusercontent.com/749374/234833879-ca0d1c6c-4849-429a-a7da-6f3f7729c1f6.png">

There's probably a way to make it work with `module.exports`, but couldn't find one, so if a JSDoc expert has any suggestion to make please let me know ;)

The intermediate config variable simply looked like a good-enough workaround

---

Note in the future, with the need to use ES modules in config, more and more sites will probably need to move to another form of config declaration using an async function, like this one:

```tsx
/** @returns {Promise<import('@docusaurus/types').Config>} */
module.exports = async function createConfigAsync() {
  return {
    // your config here
  };
};
```

TLDR: there's not a single way/syntax to declare your config. You have the full power of Node.js, as long as you return an object that matches the shape we expect (or a function that creates that object).

**Comment by thadguidry** [comment]
OMG so helpful @slorber  you solved the puzzle basically.  And yes, I wanted to keep the autocompletion and TypeScript checking working and grieved that I lost it!

So then for this issue let's do a few things to improve easily! :

- [ ] Add a `// comment` in the shipping `docusaurus.config.js` at the top such as: 
  - `// NOTE:  We use 'const config =' as simply an easy way to help give TypeScript hints for your IDE to work with this config file.  There is not a single way/syntax to declare your config within it, but note that it gets used much much further down this file, where, 'module.exports =' is the syntax that expects an object (or a function that creates that object, your choice) that returns the expected shape of a docusaurus-config See also: https://docusaurus.io/docs/api/docusaurus-config#overview`

- [ ] Update the https://docusaurus.io/docs/configuration page to state the above, preferably in a new section called '#about-the-default-docusaurus.config.js' that is placed last after '#custom-configurations', which will:
  - [ ] Also add @slorber 's image shown above to show that either syntax works, but only one syntax shows an error. Super F'ing helpful!
  - [ ] Also add @slober 's 'async' example into the page under '#custom-configurations'

**Comment by slorber** [comment]
> Add a // comment in the shipping docusaurus.config.js at the top such as:

If we add a comment in the init template, I'd like the comment to be very light (2 lines max) and link to the doc page for more details.

----


We have 2 pages:
- [docusaurus.io/docs/api/docusaurus-config#overview](https://docusaurus.io/docs/api/docusaurus-config#overview): ref doc
- [docusaurus.io/docs/configuration](https://docusaurus.io/docs/configuration): guide doc

IMHO:
- Both pages should link to each other at the top in a very visible way.
- Mentioning equivalent docs format (including the `const config` thing is more suited to live in the guide page
- Agree that the async config section in a summary/details is not visible enough. We should add these in expended form to the guide page, but still mention it's possible to use async in the API ref page, and add another link to the guide page.

In general, if you want to add long/verbose documentation on some subject, I prefer it to be in the guides part

**Comment by thadguidry** [comment]
@slorber Looks like your normal convention of API referencing is with an blue INFO admonition typically at the top.  I saw this across about 5+ pages throughout the site.

![image](https://user-images.githubusercontent.com/986438/235282837-52b18b03-ad71-4a98-b5c2-92136ed2c1b7.png)

**Comment by thadguidry** [comment]
Dev Preview of Configuration after update:
![image](https://user-images.githubusercontent.com/986438/235284859-21d75746-9b16-4304-8584-fd736e138953.png)


Dev Preview of API after update:
![image](https://user-images.githubusercontent.com/986438/235284894-4414a481-901c-46c7-aa9e-dc5e4c74632b.png)
